src/core/network/qgsnetworkdiskcache.cpp

```diff
@@ -111,3 +114,75 @@ void QgsNetworkDiskCache::clear()
   const QMutexLocker lock( &sDiskCacheMutex );
   return sDiskCache.clear();
 }
+
+qint64 QgsNetworkDiskCache::smartCacheSize( const QString &cacheDir )
+{
+  static qint64 cacheSize = 0;
+  static std::once_flag initialized;
+  std::call_once( initialized, [ = ]
+  {
+    std::function<qint64( const QString & )> dirSize;
+    dirSize = [&dirSize]( const QString & dirPath ) -> qint64
+    {
+      qint64 size = 0;
+      QDir dir( dirPath );
+
+      const QStringList filePaths = dir.entryList( QDir::Files | QDir::System | QDir::Hidden );
+      for ( const QString &filePath : filePaths )
+      {
+        QFileInfo fi( dir, filePath );
+        size += fi.size();
+      }
+
+      const QStringList childDirPaths = dir.entryList( QDir::Dirs | QDir::NoDotAndDotDot | QDir::System | QDir::Hidden );
```

Comment:
to avoid potential infinite recursion if someone nasty would create symbolic links with a recursion:

```suggestion
      const QStringList childDirPaths = dir.entryList( QDir::Dirs | QDir::NoDotAndDotDot | QDir::System | QDir::Hidden | QDir::NoSymLinks );
```